### PR TITLE
Remove unused are hosts set check

### PR DESCRIPTION
**Pull request by jm96441n**
### Description

Follow up PR from https://github.com/hashicorp/consul/pull/16677#discussion_r1142437422, this handles removing the check for `AreHostsSet`

### Testing & Reproduction steps

N/A

### Links

https://github.com/hashicorp/consul/pull/16677#discussion_r1142437422

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

**Review by andrewstucki — COMMENTED** [review]
@jm96441n can you also just go ahead and remove the field from the struct itself since this makes it completely unused? Thanks!

File: agent/proxycfg/snapshot.go [hunk 1]
```diff
@@ -706,12 +708,11 @@ type configSnapshotAPIGateway struct {
 	BoundGatewayConfig       *structs.BoundAPIGatewayConfigEntry
 
 	// Hosts is the list of extra host entries to add to our leaf cert's DNS SANs
-	Hosts       []string
-	AreHostsSet bool
+	Hosts []string
```
Comment (andrewstucki) [inline agent/proxycfg/snapshot.go:711]: Given the above docstring and the fact that we don't actually add any SANs to the leaf cert, I think this is unused as well, no? Can you confirm and drop as well?

Comment (jm96441n) [reply agent/proxycfg/snapshot.go:711]: can do

File: agent/proxycfg/api_gateway.go [hunk 2]
```diff
@@ -393,7 +393,6 @@ func (h *handlerAPIGateway) handleRouteConfigUpdate(ctx context.Context, u Updat
 	}
 	snap.APIGateway.UpstreamsSet.set(ref, seenUpstreamIDs)
 	// snap.APIGateway.Hosts = TODO
```
Comment (nathancoleman) [inline agent/proxycfg/api_gateway.go:395]:
```suggestion
```

**Review by nathancoleman — APPROVED** [review]

**Pull request merged**